.github/workflows/tests.yml

```diff
@@ -258,8 +257,124 @@ jobs:
         env:
           VS90COMNTOOLS: C:\Program Files (x86)\Common Files\Microsoft\Visual C++ for Python\9.0\VC\bin
           LIB:
-      - uses: actions/upload-artifact@v2
+
+      - name: Upload Pytest Replay
+        uses: actions/upload-artifact@v2
         with:
           path: ${{ github.workspace }}/pytest-replay
           name: Win-${{ env.CONDA_VERSION }}-Python${{ matrix.python-version }}-${{ env. parameters.serial_or_parallel  }}
         if: always()
+
+
+  macOS:
+    runs-on: macOS-10.14
+    defaults:
+      run:
+        shell: bash -l {0}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.7', '3.8']
+        conda-version: [release]
+        test-type: [serial, parallel]
+        include:
+          - python-version: '3.9'
+            conda-version: canary
+            test-type: serial
+          - python-version: '3.9'
+            conda-version: canary
+            test-type: parallel
```

Comment:
Until we have more self-hosted GitHub Actions runners via @rhoule-anaconda's work, we need to limit the amount of Python version to one [like in the conda repo](https://github.com/conda/conda/blob/7651023ca56c4bcdf200090e75ae99f62124524c/.github/workflows/ci.yml#L110) since the number of action runners is shared in the whole conda organization.